### ISSUE TITLE
Type error in BatchPlant_StandardWater

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1560]
**Reported by perost on 1 Sep 2014 08:58 UTC**
In Modelica.Fluid.Examples.AST_BatchPlant BatchPlant_StandardWater has an instance of BaseClasses.TankWith3InletOutletArraysWithEvaporatorCondensor called B5. That instance has a component mXi that's defined like this:
```
  Real[Medium.nXi] mXi(quantity = Medium.substanceNames, each unit = "kg");
```
Medium.substanceNames comes from StandardWater->WaterIF97_ph->WaterIF97_base and is set to {"water"}. Medium.nXi comes from PartialMedium, and is 0 because fixedX = true in StandardWater (inherited from PartialPureSubstance). So with these values the declaration becomes:
```
  Real[0] mXi(quantity = {"water"}, each unit = "kg");
```
which is a type error (Dymola doesn't check this).

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1560

**Comment by modelica-trac-importer** [comment]
**Modified by sjoelund&#x2e;se on 1 Sep 2014 09:59 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by fcasella on 3 Jan 2015 18:51 UTC**
Fixed in 2fa58570d3d5eb65bed70c424f3ddb487047cf1e (MSL 3.2.1 build 3) and 924dd4f72af89619cff22e4c6df8815e98e95f6e (trunk)
